src/operator/nn/pool.cuh

```diff
@@ -311,7 +322,9 @@ __global__ void pool_sum_3d_gpu_kernel(const int nthreads, const DType* in_data,
         }
       }
     }
-    out_data[index] = a_root_p<DType, p>::Map(sum);
+    out_data[index] = (pool_size == 0) ?
+                      DType(0.0f / pool_size) :
```

Comment:
division by 0?

Reply:
We want to create an artificial NaN here to ensure consistency with cudnn.

Reply:
would it be better to use nan constants in `math/limits`? performing this division produces nan through interrupt which causes slow down.

Reply:
You cannot use the std thing in gpu kernels, as they are __host__ functions while this kernel here is a __global__ function.

Reply:
how about [nan](https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__DOUBLE.html#group__CUDA__MATH__DOUBLE_1g6df5511321a5ac0dfe22389b728a8a9f) and  [nanf](https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__SINGLE.html#group__CUDA__MATH__SINGLE_1g372c640f910303dc4a7f17ce684322c5)?

Reply:
What about fp16?

Reply:
Why would pool_size be 0? Shouldn't that be invalid?

Reply:
When your pad is wider than kernel size, you could get that pool_size when count_include_pad is False.

Reply:
For example in the 3d test cases, there's one with (20, 20, 20) input, (4,5,3) kernel, and (2,3,3) pad. So on the last dim you could get 0 valid width and your whole pool_size will be 0.

Reply:
DType cast should have already handled it for half_t and half2_t, or did I miss something?

Reply:
@szha okay then I'll just use those functions to generate NaN.